lib/ansible/executor/task_executor.py

```diff
@@ -642,14 +651,23 @@ def _execute(self, variables=None):
         for attempt in xrange(1, retries + 1):
             display.debug("running the handler")
             try:
+                if self._task.timeout:
+                    old_sig = signal.signal(signal.SIGALRM, task_timeout)
+                    signal.alarm(self._task.timeout)
                 result = self._handler.run(task_vars=variables)
             except AnsibleActionSkip as e:
                 return dict(skipped=True, msg=to_text(e))
             except AnsibleActionFail as e:
                 return dict(failed=True, msg=to_text(e))
             except AnsibleConnectionFailure as e:
                 return dict(unreachable=True, msg=to_text(e))
+            except TimeoutError as e:
```

Comment:
I would call this `AnsibleTimeoutError` (or even `AnsibleTaskTimeoutError`) for consistency with the other custom ansible exceptions

Reply:
the difference is that this is not derived from AnsibleError and should not be 'capturable' as such.